glamox/src/Components/CartDrawer.tsx

```diff
@@ -1,280 +0,0 @@
-  //     setCart(res.data)
-  //   })
-  //   .catch((err)=>{
-  //     console.error(err)
-  //   })
-  // }
-
-  const handleRemove = (id: number) => {
-    //setCart((prev:any) => prev.filter((el:any, i:any) => i != id));
-    // axios.delete(`https://fine-puce-bison-cap.cyclic.app/cart/${id}`).then((r)=>{
-    // let x = axios.get('https://fine-puce-bison-cap.cyclic.app/cart').then((r)=>{
-    //   setCart(x);
-    // }).catch((e)=>console.log(e))
-    //   console.log(r.data)
-    // }).catch((e)=> console.log(e));
-    // dispatch(removeCart(id))
-  };
-
-  const handleQuantity = (index: number, quan: any) => {
-    let newquan = Number(quan.target.value);
-    console.log(quan.target.value);
-
-    // setCart((prev:any) =>
-    //   prev?.map((el:any, i:any) => {
-    //     return i == index ? { ...el, qty: newquan } : { ...el };
-    //   })
-    // );
-  };
-
-  const handlePrice = (index: number) => {
-    for (let i = 0; i < cart.length; i++) {
-      if (i == index) return cart[i].altprice();
-    }
-  };
-
-  useEffect(() => {
-    let sum = 0;
-    let mrp;
-    cart.map((item: any) => (sum += item.price * item.qty));
-    dispatch({ type: "total", payload: sum });
-    mrp = Math.round(sum + (12 / 100) * sum);
-    dispatch({ type: "mrp", payload: mrp });
-    dispatch({ type: "discount", payload: mrp - sum });
-  }, [cart]);
-
-  useEffect(() => {
-    //   fetchcartdata();
-    setCart(myCart);
-  }, [myCart]);
-
-  return (
-    <>
-      <Box
-        ref={btnRef}
-        border={"0px"}
-        p={"2"}
-        borderRadius={"full"}
-        onClick={onOpen}
-        _hover={{ cursor: "pointer", backgroundColor: "teal", color: "white" }}
-      >
-        <Flex border={"0px"} w={"130%"}>
-          <BsBag />
-          <Text
-            fontSize={"x-small"}
-            border={"0px"}
-            borderRadius={"50%"}
-            p={"0% 10%"}
-            bg={"red.500"}
-            color={'white'}
-          >
-            {cart.length}
-          </Text>
-        </Flex>
-      </Box>
-      <Drawer
-        isOpen={isOpen}
-        placement="right"
-        onClose={onClose}
-        finalFocusRef={btnRef}
-        size={"md"}
-      >
-        <DrawerOverlay />
-        <DrawerContent>
-          <DrawerCloseButton />
-          <DrawerHeader>Bag</DrawerHeader>
-          <hr />
-          <DrawerBody>
-            {/* -------------------------Cart Items------------------------------------------ */}
-            <VStack>
-              {cart?.map((item: any, index: any) => (
-                <CartItem
-                  key={item.id}
-                  {...item}
-                  handleRemove={handleRemove}
-                  index={index}
-                  handleQuantity={handleQuantity}
-                  handlePrice={handlePrice}
-                />
-              ))}
-            </VStack>
-            {/* -----------------------Cart Price Details------------------------------------ */}
```

Comment:
We can move the Cart price Details section to a sub component and import it here to make this component more readable